cookbooks/aws-parallelcluster-environment/recipes/init/config_default_user_home.rb

```diff
@@ -44,18 +44,37 @@
   EOH
 end
 
-# move the cluster user's default home directory
+# Move the cluster user's default home directory
+# This script performs the following actions:
+# 1. Creates the new local home directory for the cluster user if it doesn't already exist.
+# 2. Copies the data from the temporary backup directory (/tmp/cluster_user_home) to the new local home directory.
+# 3. Updates the cluster user's home directory path to the new local home directory.
+# 4. Changes the ownership of the new local home directory to the cluster user.
+# 5. Verifies data integrity by comparing the temporary backup directory and the new local home directory.
+# 6. If the data integrity check passes, it removes both the temporary backup directory and the original home directory.
+# 7. If the data integrity check fails, it outputs an error message, restart sshd service and exits with an error code.
+# 8. Unhappy Path Manually test passed successfully:
+#    Simulate a failure in the data integrity check by creating expected `node['cluster']['cluster_user_local_home']`
+#    Modify files in the expected directory to differ from the original.
+#    Manually run `config_default_user_home` recipe.
+#    Verified that the script fails and outputs the correct error message in chef-client.log.
 bash "Move #{node['cluster']['cluster_user_home']}" do
   user 'root'
   group 'root'
   code <<-EOH
-    set -e
```

Comment:
By removing this setting, we allow the process to continue even there are failures in the mkdir, rsync, usermod, chown, etc.  Does this behave properly in those cases?

It might be better in this case to move the diff check to a separate bash resource.  This might also make it easier to unit test

Reply:
Great catch and advice, thank you! Have corrected it.